fmlloader/src/main/java/net/minecraftforge/fml/loading/targets/CommonLaunchHandler.java

```diff
@@ -72,6 +73,12 @@ record ExplodedModPath(String modid, Path path) {}
                 .map(splitString -> new ExplodedModPath(splitString.length == 1 ? "defaultmodid" : splitString[0], Paths.get(splitString[splitString.length - 1])))
                 .collect(Collectors.groupingBy(ExplodedModPath::modid, Collectors.mapping(ExplodedModPath::path, Collectors.toList())));
 
+        // We need to REVERSE the passed in mod classes so that SJH takes it properly.
+        // The last path passed into SecureJar#from is the "main" path used to grab the manifest.
+        // ForgeGradle always passed in resources,classes with the expectation in 1.16 or lower that the resources would be checked first.
+        // So in 1.17+ FML/SJH, we need to reverse it.
+        modClassPaths.forEach((k, v) -> Collections.reverse(v));
```

Comment:
So the problem isn't that things are in the wrong order its that things can quite often be in ANY order.
So the best option is the trust the order that exists, but put the one that specified the manifest files at the first one.
So instead of reverse. Sort.
Would a possible option be to copy over the manifest to every stream thats opened?
Or, since this is a read only file system. What we could do is make a fake input stream provider that sorts the entries/lists all files as if they were zips.

Reply:
This is removed/fixed as of [`6d9030b` (#7976)](https://github.com/MinecraftForge/MinecraftForge/pull/7976/commits/6d9030be92921e297c3859cf7145ed99f904fd4a) because SJH 0.9.51 and up properly checks ALL paths to find a manifest instead of just the last one.